next/nextjs-project/lib/mongodb.ts

```
@@ -0,0 +1,63 @@
+import mongoose from "mongoose";
+
+// Define the connection cache type
+type MongooseCache = {
+    conn: typeof mongoose | null;
+    promise: Promise<typeof mongoose> | null;
+};
+
+// Extend the global object to include our mongoose cache
+declare global {
+    // eslint-disable-next-line no-var
+    var mongoose: MongooseCache | undefined;
```

Biome (2.1.2)

[error] Shouldn't redeclare 'mongoose'. Consider to delete it or rename it.

'mongoose' is defined here:


(lint/suspicious/noRedeclare)